src/main/java/seedu/address/logic/commands/DeleteRecurringSessionCommand.java

```diff
@@ -0,0 +1,90 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INDEX;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TIME;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.session.RecurringSession;
+import seedu.address.model.session.Session;
+import seedu.address.model.session.SessionDate;
+import seedu.address.model.student.Name;
+import seedu.address.model.student.Student;
+
+/**
+ * Deletes a session from a recurring session identified using it's displayed index from the address book, date, and
+ * time of the session to be deleted.
+ */
+public class DeleteRecurringSessionCommand extends Command {
```

Comment:
Nice! Seems to work well for recurring sessions with multiple dates. However, I tried to use delete_rec_session on a recurring session with only 1 date and it returns an exception, maybe you want to look into that.

![image](https://user-images.githubusercontent.com/24240408/113134420-56bb7a00-9253-11eb-9eee-70fe0c962a34.png)

Reply:
Good spot! Forgot about this, will add the corner case in. Thanks!! 😁